src/scope.rs

```diff
@@ -11,7 +11,6 @@ use crate::{named::ProgExt, ProgNode};
 #[derive(Debug, Clone)]
 pub struct GlobalScope {
     variables: Vec<Vec<Pattern>>,
-    witnesses: Vec<Vec<WitnessName>>,
```

Comment:
Witness names are unscoped but the variables that hold them _are_ scoped. `witness("name")` returns a constant value like a jet `jet_version`.